W-20179825 Refactor agent network documentation (#549)
mulesoft/docs-code-builder: modules/ROOT/pages/af-define-your-agent-network-specification.adoc

= Define Your Agent Network Specification

After you create your agent network project, configure `agent-network.yaml` and `exchange.json` to reflect the structure of your network. 

== Before You Begin 

Make sure you review the xref:af-get-started.adoc#before-you-begin[prerequisites].

[TIP]

> To help you identify agents that are also brokers on the Anypoint Code Builder canvas, consider appending "Broker" to the end of the name.

Reviewer: Nice. It would also be nice to have an example name, but it's not that important.

> == Define a Network Using the MuleSoft Dev Agent

Reviewer: ```suggestion
== Define a Network Using MuleSoft Dev Agent
```

. In the Anypoint Code Builder activity bar, click the agent icon image:af-acb-dev-agent-icon.png["",18,18].
. Give the agent information about your agent network, including the brokers, agents, MCP servers, and LLMs you want to connect.

To get started try one of these suggested prompts.

* Help me build an agent network configuration for my Employee Onboarding project.
* Revise my Employee Onboarding agent network project. Use OpenAI for the LLM.
* Add a new broker for employee offboarding to my HR agent network.
* Add a new skill to Offboarding broker.

== Define a Network Using Anypoint Code Builder or IDE

If you don't want to use MuleSoft Dev Agent, use Anypoint Code Builder or your IDE to edit the `agent-network.yaml` and `exchange.json` files and define your agent network and authentication.

To understand sections of the project files and expected values, see xref:anypoint-code-builder::af-project-files.adoc[Agent Network Project File Reference]. The agent-network.yaml file can contain definitions for one or more brokers.

> If your agent network references Anypoint Exchange assets, you need to use the asset IDs in `exchange.json` to add references in the `agent-network.yaml` file. For more information, see Add Exchange Assets to Your Agent Network Project.

Reviewer: Add a link for this? 
For more information, see Add Exchange Assets to Your Agent Network Project.

Use auto-completion menus in Anypoint Code Builder to speed your development. For more information, see xref:anypoint-code-builder::start-discover-ui.adoc#use-autocomplete[Use Auto-Completion Menus].

== Add Exchange Assets to Your Agent Network Project

> If you have existing Exchange assets you want to leverage in your agent network, add them to the dependencies attribute in `exchange.json` in your project. After you add assets, edit agent-network.yaml to indicate which brokers use those assets.

Reviewer: ```suggestion
If you have existing Exchange assets you want to use in your agent network, add them to the dependencies attribute in `exchange.json` in your project. After you add assets, edit the `agent-network.yaml` file to indicate which brokers use those assets.
```

> === Add Assets Using the MuleSoft Dev Agent

Reviewer: ```suggestion
=== Add Assets Using MuleSoft Dev Agent
```

> . In the Anypoint Code Builder activity bar, click the agent icon image:image3[].

Reviewer: Is this image showing up correctly in your local build?

. Tell the agent that you want to add Exchange assets to your project. MuleSoft Dev Agent does the rest.

To get started try one of these suggested prompts.

* Add tools for background check processing in my Employee Onboarding project.

> * To my Employee Onboarding agent network project, add Talent Pool MCP server.

Reviewer: ```suggestion
*  Add a Talent Pool MCP server to my Employee Onboarding agent network project.
```

=== Add Assets Using Anypoint Code Builder

. If you're not logged in already, log in to your Anypoint Platform account.
. In your Anypoint Code Builder project, choose one of the following:
  * In Explorer, right-click a project file and select *Add Exchange Assets to Agent Network Project*.
  * In the Command Palette, run this command: *MuleSoft: Add Agent Network Assets to Agent Network Project*.

> . In Add Exchange Assets to Project, enter information about the assets you want to add.

Reviewer: ```suggestion
. In Add Exchange Assets to Project, enter information about the assets to add.
```

> . Select *Add to Project*. Assets are added to the dependencies attribute in the `exchange.json` file in your project.

Reviewer: ```suggestion
. Select *Add to Project*. 
+
Assets are added to the `dependencies` attribute in the `exchange.json` file in your project.
```

> . Edit agent-network.yml L to indicate which brokers use those assets.

Reviewer: ```suggestion
. Edit the `agent-network.yml` file to indicate which brokers use those assets.
```

>   If you added a dependency that doesn't belong to the same business group as your agent network, then in agent-network.yaml, specify the namespace property at the same level as name. Provide the groupId (business group ID) for the business group that the dependency belongs to. The groupId value is the same as the groupId value of the corresponding dependency in exchange.json.

Reviewer: ```suggestion
  If you added a dependency that doesn't belong to the same business group as your agent network, then in `agent-network.yaml`, specify the `namespace` property at the same level as name. Provide the `groupId` (business group ID) for the business group that the dependency belongs to. The `groupId` value is the same as the `groupId` value of the corresponding dependency in `exchange.json`.
```

> If you don't provide a namespace value, the same groupId as the agent network project is used.

Reviewer: ```suggestion
If you don't provide a namespace value, the same `groupId` as the agent network project is used.
```

> After you add dependencies, they're available as values in auto-completion menus in Anypoint Code Builder. For example, after you add test-agent to exchange.json, the value test-agent is available in auto-completion menus in the code editor in agent-network.yaml when you reference an agent.

Reviewer: ```suggestion
After you add dependencies, they're available as values in auto-completion menus in Anypoint Code Builder. For example, after you add `test-agent` to `exchange.json`, the value `test-agent` is available in auto-completion menus in the code editor in `agent-network.yaml` when you reference an agent.
```
